src/java.base/share/classes/jdk/internal/classfile/TypeAnnotation.java

```diff
@@ -0,0 +1,652 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.classfile;
+
+import java.lang.constant.ClassDesc;
+import java.util.List;
+import java.util.Set;
+
+import jdk.internal.classfile.attribute.RuntimeInvisibleTypeAnnotationsAttribute;
+import jdk.internal.classfile.attribute.RuntimeVisibleTypeAnnotationsAttribute;
+import jdk.internal.classfile.constantpool.Utf8Entry;
+import jdk.internal.classfile.impl.TargetInfoImpl;
+import jdk.internal.classfile.impl.UnboundAttribute;
+
+import static jdk.internal.classfile.Classfile.TAT_CAST;
+import static jdk.internal.classfile.Classfile.TAT_CLASS_EXTENDS;
+import static jdk.internal.classfile.Classfile.TAT_CLASS_TYPE_PARAMETER;
+import static jdk.internal.classfile.Classfile.TAT_CLASS_TYPE_PARAMETER_BOUND;
+import static jdk.internal.classfile.Classfile.TAT_CONSTRUCTOR_INVOCATION_TYPE_ARGUMENT;
+import static jdk.internal.classfile.Classfile.TAT_CONSTRUCTOR_REFERENCE;
+import static jdk.internal.classfile.Classfile.TAT_CONSTRUCTOR_REFERENCE_TYPE_ARGUMENT;
+import static jdk.internal.classfile.Classfile.TAT_EXCEPTION_PARAMETER;
+import static jdk.internal.classfile.Classfile.TAT_FIELD;
+import static jdk.internal.classfile.Classfile.TAT_INSTANCEOF;
+import static jdk.internal.classfile.Classfile.TAT_LOCAL_VARIABLE;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_FORMAL_PARAMETER;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_INVOCATION_TYPE_ARGUMENT;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_RECEIVER;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_REFERENCE;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_REFERENCE_TYPE_ARGUMENT;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_RETURN;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_TYPE_PARAMETER;
+import static jdk.internal.classfile.Classfile.TAT_METHOD_TYPE_PARAMETER_BOUND;
+import static jdk.internal.classfile.Classfile.TAT_NEW;
+import static jdk.internal.classfile.Classfile.TAT_RESOURCE_VARIABLE;
+import static jdk.internal.classfile.Classfile.TAT_THROWS;
+import jdk.internal.classfile.impl.TemporaryConstantPool;
+
+/**
+ * Models an annotation on a type use.
+ *
+ * @see RuntimeVisibleTypeAnnotationsAttribute
+ * @see RuntimeInvisibleTypeAnnotationsAttribute
+ */
+public sealed interface TypeAnnotation
+        extends Annotation
+        permits UnboundAttribute.UnboundTypeAnnotation {
+
+    /**
+     * The kind of target on which the annotation appears.
+     */
+    public enum TargetType {
```

Comment:
There is only one TargetInfo for all TargetTypes, so instead of 22 sub-interfaces of TargetInfo, the instance of TargetType enum is hold in TargetInfo.

Reply:
Ok, I see that now - for some reason the IDE could not find the usage... thanks